### ISSUE TITLE
Add possibility to navigate in folder hierarchie

### ISSUE THREAD

**Issue by godfuture** [nextcloud/android#2644]
### Actual behaviour
Navigating in deep folder structures is a bit frustrating, because I keep clicking the "back" (arrow) button. Clickling one of the navigation element in the left swipe menu sends me strictly to the root dir and is not really a help to reach folders in between.

### Expected behaviour
The current folder title is a button. When I click the button, it shows me the folder path / structure as a list representing the folder hierarchie. Alternatively longpressing the backbutton is fine too, but the folder title as button would be faster in usability.
 
### Steps to reproduce
1. Create a deep folder structure on server
2. Navigate to the last element in the hierarchie
3. Try to navigate back to a folder of choice

### Environment data
Android version: 7.1.2

Device model: BQ X5P

Stock or customized system: LineageOS

Nextcloud app version: 3.1.0

Nextcloud server version: 13.0.2

#### Web server error log
```
-
```

#### Nextcloud log (data/nextcloud.log)
```
-
```

**Comment by AndyScherzinger** [comment]
Thanks for the report @godfuture 

What do you think @mario @tobiasKaminsky @nextcloud/designers / @jancborchardt It would basically be the same thing we already have for the local file list (used for manula upload file selection).

**Comment by tobiasKaminsky** [comment]
I am fine with this, but then we can maybe enhance it a bit and make it looks nicer?

Here is how others are doing it:

Files app:
![2018-05-30-083812](https://user-images.githubusercontent.com/5836855/40703236-d614fa24-63e4-11e8-9189-53c86a85d692.png) ![2018-05-30-083806](https://user-images.githubusercontent.com/5836855/40703237-d639cfe8-63e4-11e8-90ff-b773abf0f73f.png)

Amaze
![2018-05-30-083616](https://user-images.githubusercontent.com/5836855/40703152-8ce69f56-63e4-11e8-96b3-7525a926e9de.png) ![2018-05-30-083619](https://user-images.githubusercontent.com/5836855/40703153-8d062cc2-63e4-11e8-9ca5-ad9ef73a0b7b.png)

KDE dolphin (file browser)
![image](https://user-images.githubusercontent.com/5836855/40703102-6fa2ebd4-63e4-11e8-8e84-cb176370861e.png)

**Comment by AndyScherzinger** [comment]
sure thing!

**Comment by jancborchardt** [comment]
Good stuff! The first example of how the files app does it is good! I'd say though the dropdown arrow is not needed there as the function is kind of an expert shortcut.

**Comment by tobiasKaminsky** [comment]
Then it is close to what we have now if you upload a file:
![2018-06-01-080215](https://user-images.githubusercontent.com/5836855/40823823-1f06cb8e-6572-11e8-99bf-41350209c95a.png)

Then I'd say that we stay with it, but invert black/white?

**Comment by AndyScherzinger** [comment]
Sounds good to me, the triangle is part of the standard spinner so that'll still be there ;) :+1:

**Comment by jancborchardt** [comment]
@tobiasKaminsky yup, exactly. And yes, since the general theme is white, that menu should also be white.

@AndyScherzinger aah damn … well, can’t have everything. :D

**Comment by AndyScherzinger** [comment]
Well, for the long run we should change it to something like Amaze File Explorer since the tinting for the spinner triangle now needs to use an internal method of the support library which is something that might be an issue in the future but for the short term we can go with this and then move on to our own solution :)
